src/app/home/reminder.service.ts

```diff
@@ -1,105 +1,107 @@
+    const docRefId = snapshot2.docs[0].id;
+    return docRefId
   }
 
-  updateReminder(key, newValue) {
-    this.storage.set(key, newValue);
-    this.getAllReminders();
+  async deleteReminder(item) { //delete item
+    const userId = await this.getUserId()
+    const ToDoId = await this.getToDoItem(userId, item)
+    await deleteDoc(doc(db, "users", userId, "reminders", ToDoId));
   }
 
-  getReminder(key) {
-    return this.storage.get(key);
+  async getAllReminders() { //get all reminders
+    const docRefId = await this.getUserId()
+
+    let array = []
+    //gets sub collection
+    const subColRef = await getDocs(collection(db, "users", docRefId, "reminders"));
+    subColRef.forEach(d => {
+      array.push(d.data())
+    })
+    return array;
   }
 
-  getAllReminders() {
-    // eslint-disable-next-line prefer-const
-    let reminders: any = [];
-    this.storage.forEach((value, key, index) => {
-      reminders.push({ key: value, value: key });
-    });
-    return reminders;
-    // return reminders;
+  async setCompleted(item) { //sets reminder as completed
+    const userId = await this.getUserId()
+    const ToDoId = await this.getToDoItem(userId, item)
+    await updateDoc(doc(db, "users", userId, "reminders", ToDoId), item);
   }
 
-  setCompleted(key, newValue){
-    this.storage.set(key, newValue);
-    this.getAllReminders();
+  async updateReminder(item){
+      const userId = await this.getUserId()
+      const ToDoId = await this.getToDoItem(userId, item)
+      await updateDoc(doc(db, "users", userId, "reminders", ToDoId), item);
   }
 
-  getUncompletedReminders(){
-    let reminders: any = [];
-    this.storage.forEach((value, key, index) => {
-      if(value.isCompleted === false){
-      reminders.push({ key: value, value: key });
-    }
-    });
-    // console.log(reminders)
-    return reminders;
+  async getUncompletedReminders() {
+    const docRefId = await this.getUserId()
+
+
+    let array = []
+    //gets sub collection
+    const q2 = await query(collection(db, "users", docRefId, "reminders"), where("isCompleted", "==", false));
+    const snapshotUncompleted = await getDocs(q2);
+    snapshotUncompleted.forEach(d => {
+      array.push(d.data())
+    })
+    return array;
   }
 
-  getCompletedReminders(){
-    let reminders: any = [];
-    this.storage.forEach((value, key, index) => {
-      if(value.isCompleted === true){
-      reminders.push({ key: value, value: key });
-    }
-    });
-    return reminders;
+  async getCompletedReminders() {
+    const docRefId = await this.getUserId()
+
+    let array = []
+    //gets sub collection
+    const q2 = await query(collection(db, "users", docRefId, "reminders"), where("isCompleted", "==", true));
+    const snapshotUncompleted = await getDocs(q2);
+    snapshotUncompleted.forEach(d => {
+      array.push(d.data())
+    })
+    return array;
   }
 
-  getReminderCount(){
+  getReminderCount() {
     this.storage.length().then(result =>
```

Comment:
@gisellen Would there still be a use for ionic storage in this file ?  it seems like in this file we have changed to using firebase and we can use its functionalities to access the data in similar ways. If we dont need it i would recommend removing the lines associated to ionic storage.

Reply:
@Brandon-Torres No, all the ionic storage stuff should be gone. I just haven't been able to touch upon the other functions besides add/update/delete